src/Cli/Microsoft.TemplateEngine.Cli/Commands/create/TemplateCommandArgs.cs

```diff
@@ -18,7 +22,19 @@ public TemplateCommandArgs(TemplateCommand command, BaseCommand parentCommand, P
             ParentCommand = parentCommand ?? throw new ArgumentNullException(nameof(parentCommand));
             RootCommand = GetRootCommand(parentCommand);
 
-            Name = parseResult.GetValueForOptionOrNull(SharedOptions.NameOption);
+            // Prefer the name argument over the hidden name option (which is kept for backward compatibility).
+            var nameArgument = parseResult.GetValue(SharedOptions.NameArgument);
+            if (nameArgument == NameDefaultSentinel)
+            {
+                nameArgument = null;
+            }
+            var nameOption = parseResult.GetValueForOptionOrNull(SharedOptions.NameOption);
+            if (nameArgument is not null && nameOption is not null)
+            {
+                throw new InvalidOperationException($"Name argument '{nameArgument}' and name option (--name) '{nameOption}' have both been provided. Only one may be provided at a time.");
```

Comment:
Is this the right way to error here? I've seen some throws like this, but it would also mean that these situations are unlocalized.

Reply:
I've added a validator for this situation. When are the validators executed?